src/main/java/org/junitpioneer/jupiter/DisableIfTestFailsExtension.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.TestExecutionExceptionHandler;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+class DisableIfTestFailsExtension implements TestExecutionExceptionHandler, ExecutionCondition {
+
+	/*
+	 * Basic approach:
+	 *  - in `handleTestExecutionException`: if need to deactivate other tests, add that info to the store
+	 *  - in `evaluateExecutionCondition`: check store for that information
+	 *
+	 * Because the test method that failed and the ones that need to be disabled are different methods,
+	 * the information to disable can't be in a store that belongs to any specific test method. Instead
+	 * add it to the store that belongs to the container where the extension is applied.
+	 *
+	 * Setting the information needs to be thread safe, so only positive results (i.e. tests must be disabled)
+	 * will be set. The easiest way to do that is to simply use absence/presence of a key in the store
+	 * as indicator, which means the specific value doesn't.
+	 */
+
+	private static final Namespace NAMESPACE = Namespace.create(DisableIfTestFailsExtension.class);
+	private static final String DISABLED_KEY = "DISABLED_KEY";
+	private static final String DISABLED_VALUE = "";
+
+	@Override
+	public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+		boolean disabled = context.getStore(NAMESPACE).get(DISABLED_KEY) != null;
+		if (disabled)
+			return ConditionEvaluationResult.disabled("Another test threw one of the specified exceptions.");
+		else
+			return ConditionEvaluationResult.enabled("No test threw one of the specified exceptions (yet).");
+	}
```

Comment:
Why only disabled if another test threw an exception? Why not disable if another test fails for any reason?

Reply:
That's not a bad point, I have to think about that. First impulse: "Because this exists to deactivate tests that make "no sense to run" because some infrastructure failed." But this might not actually be a good argument. 😊

Reply:
You're right that this scenario needs to be considered. I think assertion errors should be special cased and it should be possible to configure how the extension treats them. I got a work-in-progress commit lying around locally.

Reply:
Implemented both, thanks for spotting that!